### ISSUE TITLE
naive datetimes exported to json not valid with generate json schema

### ISSUE THREAD

**Issue by languitar** [pydantic/pydantic#2460]
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.7.3
            pydantic compiled: True
                 install path: /home/languitar/src/api-files/.venv/lib/python3.9/site-packages/pydantic
               python version: 3.9.2 (default, Feb 20 2021, 18:40:11)  [GCC 10.2.0]
                     platform: Linux-5.10.19-1-lts-x86_64-with-glibc2.33
     optional deps. installed: ['typing-extensions', 'email-validator']
```

When constructing a data type with a `datetime` field, the default JSON Schema generation results in a string field with format `date-time`, which is reasonable. However, it is possible to generate data when using the `json()` export method that is not compatible with this schema in case tz-naive datetime is used

```py
In [8]: from datetime import datetime

In [9]: from pydantic import BaseModel

In [10]: class Test(BaseModel):
    ...:     foo: datetime
    ...: 

In [11]: Test.schema()
Out[11]: 
{'title': 'Test',
 'type': 'object',
 'properties': {'foo': {'title': 'Foo',
   'type': 'string',
   'format': 'date-time'}},
 'required': ['foo']}

In [12]: Test(foo=datetime(2020, 1, 1, 12, 1, 1, 12345)).json()
Out[12]: '{"foo": "2020-01-01T12:01:01.012345"}'
```

The resulting string representation lacks a timezone indicator to be valid (cf. https://tools.ietf.org/html/rfc3339#section-5.6).

Of course it is not automatically possible to magically come up with a timezone for a datetime without this information. But is is also a bit surprising that still something is generated in the `json()` method that doesn't follow any valid schema. I can think of multiple solutions to make this at least more predictable for users:

1. Always fail with an exception for naive datetime instances
2. assume a default timezone (configurable) such as UTC
3. Require a default timezone to be provided in case of naive datetime instances

**Comment by Kludex** [comment]
You can now use `NaiveDatetime` and `AwareDatetime`. The `NaiveDatetime` raises validation error when a timezone info is provided, and the `AwareDatetime` raises error when the timezone info is missing.

```py
from datetime import datetime

from pydantic import BaseModel, AwareDatetime, NaiveDatetime
import pytz

class Test(BaseModel):
    foo: datetime
    bar: AwareDatetime
    por: NaiveDatetime


print(
    Test(
        foo=datetime(2020, 1, 1, 12, 1, 1, 12345),
        bar=datetime(2020, 1, 1, 12, 1, 1, 12345, tzinfo=pytz.utc),  # Remove tzinfo to fail.
        por=datetime(2020, 1, 1, 12, 1, 1, 12345), # Add tzinfo to fail.
    ).json()
)
```

**Comment by Chuck-Koverse** [comment]
@Kludex  Which version of Pydantic will these modules be available?  I get an import error with Pydantic 1.10.7.  But I would love to get these types.

**Comment by samuelcolvin** [comment]
@Kludex is using V2 which is currently available as an alpha release.
